packages/clay-date-picker/src/DayNumber.tsx

```diff
@@ -7,43 +7,81 @@ import {Keys} from '@clayui/shared';
 import classnames from 'classnames';
 import React from 'react';
 
-import {IDay, formatDate, setDate} from './Helpers';
+import {IDay, setDate} from './Helpers';
 
 interface IProps {
 	day: IDay;
-	daySelected: Date;
+	daysSelected: readonly [Date, Date];
 	disabled?: boolean;
+	range?: boolean;
 	onClick: (date: Date) => void;
 }
 
+interface IInterval {
+	start: Date;
+	end: Date;
+}
+
+function isWithinInterval(dirtyDate: Date, interval: IInterval) {
+	const time = dirtyDate.getTime();
+	const startTime = interval.start.getTime();
+	const endTime = interval.end.getTime();
+
+	if (startTime > endTime) {
+		return false;
+	}
+
+	return time >= startTime && time <= endTime;
+}
+
 const ClayDatePickerDayNumber: React.FunctionComponent<IProps> = ({
 	day,
-	daySelected,
+	daysSelected,
 	disabled,
 	onClick,
+	range,
 }) => {
 	const {date, outside} = day;
+	const [startDate, endDate] = daysSelected;
+
+	const hasEndDateSelected = date.toDateString() === endDate.toDateString();
+	const hasStartDateSelected =
+		date.toDateString() === startDate.toDateString();
 
 	const classNames = classnames(
 		'date-picker-date date-picker-calendar-item',
 		{
-			active: date.toDateString() === daySelected.toDateString(),
+			active: hasStartDateSelected || (range && hasEndDateSelected),
 			disabled: outside || disabled,
 		}
 	);
 
 	return (
-		<div className="date-picker-col">
+		<div
+			className={classnames(
+				'date-picker-col',
+				range && {
+					'c-selected':
+						!hasStartDateSelected &&
+						!hasEndDateSelected &&
+						isWithinInterval(date, {
+							end: endDate,
+							start: startDate,
+						}),
+					'c-selected c-selected-end':
+						hasEndDateSelected && !hasStartDateSelected,
+					'c-selected c-selected-start':
+						hasStartDateSelected && !hasEndDateSelected,
+				}
```

Comment:
Can we make it so only one class is declared per key? I think this would work the same...

```js
{
	'c-selected':
		hasStartDateSelected ||
		hasEndDateSelected ||
		isWithinInterval(date, {
			end: endDate,
			start: startDate,
		}),
	'c-selected-end': hasEndDateSelected && !hasStartDateSelected,
	'c-selected-start': hasStartDateSelected && !hasEndDateSelected,
}
```

Reply:
I think yes, I don't remember why it didn't go on with that.